File: LeetCode/python3/1009.py
class Solution:
    def bitwiseComplement(self, N: int) -> int:
        if N == 0:
            return 1
        m, res, = 1, 0
        while N:
            res += m * (N % 2 ^ 1)
            m <<= 1
            N >>= 1
        return res

File: LeetCode/cpp/1009.cpp
class Solution {
public:
    int bitwiseComplement(int N) {
        if (N == 0)
            return 1;
        int m = 1, res = 0;
        while (N) {
            res += m * (N % 2 ^ 1);
            N >>= 1;
            m <<= 1;
        }
        return res;
    }
};
